google/cloud/forseti/scanner/audit/retention_rules_engine.py

```diff
@@ -353,15 +377,70 @@ def find_violations_in_bucket(self, bucket):
         """Get a generator for violations.
 
         Args:
-            bucket (bucket): Find violation from the buckets.
+            bucket (Bucket): Bucket to find violation from.
         Returns:
-            Generator: All violations of the buckets breaking rules.
+            Generator: Retention violation of buckets.
         """
-
         violation_max = self.bucket_max_retention_violation(bucket)
         violation_min = self.bucket_min_retention_violation(bucket)
         return itertools.chain(violation_max, violation_min)
 
+    def table_max_retention_violation(self, table):
+        """Generate violations for bigquery table that exceeded the maximum
+        retention.
+
+        Args:
+            table (Table): Table to find violation from.
+        Yields:
+            RuleViolation: Retention violation of bigquery tables.
+        """
+        if self.max_retention is None:
+            return
+
+        table_dict = json.loads(table.data)
+        table_expiration = table_dict.get('expirationTime')
+        if not table_expiration:
+            yield self.generate_table_violation(table)
+        else:
+            table_creation = table_dict['creationTime']
+            diff = long(table_expiration) - long(table_creation)
+            if diff > self.max_retention * 24 * 3600000:
```

Comment:
Make this a constant, e.g.
`_MS_PER_DAY = 24 * 60 * 60 * 1000`

Same in the test files